lib/std/zig/string_literal.zig

```diff
@@ -280,6 +280,43 @@ pub fn parseAlloc(allocator: std.mem.Allocator, bytes: []const u8) ParseError![]
     }
 }
 
+/// Parses `bytes` as a Zig string literal and writes the result to the std.io.Writer type.
+/// Asserts `bytes` has '"' at beginning and end.
+/// Higher level API. Does not return extra info about parse errors.
+pub fn parseWrite(writer: anytype, bytes: []const u8) ParseError!void {
```

Comment:
I think that if this API returned a rich error like `parseAppend()` does having signature
```zig
pub fn parseAppend(writer: anytype, bytes: []const u8) error{OutOfMemory}!Result {
```
it would be strictly better and could replace the `parseAppend()` usage in `AstGen.zig` as well as the implementation of `parseAlloc()`. 

Note that `std.ArrayList(u8).writer()` exists, so the diff in `AstGen.zig` for example could be something like
```diff
diff --git a/src/AstGen.zig b/src/AstGen.zig
index 48e6a480f..0e2164681 100644
--- a/src/AstGen.zig
+++ b/src/AstGen.zig
@@ -9864,7 +9864,7 @@ fn parseStrLit(
 ) InnerError!void {
     const raw_string = bytes[offset..];
     var buf_managed = buf.toManaged(astgen.gpa);
-    const result = std.zig.string_literal.parseAppend(&buf_managed, raw_string);
+    const result = std.zig.string_literal.parseWrite(buf_managed.writer(), raw_string);
     buf.* = buf_managed.moveToUnmanaged();
     switch (try result) {
         .success => return,
```

If that improvement is made to this API I'd be happy to delete `parseAppend()` and remove the code duplication.